payments_service/solitude/__init__.py

```diff
@@ -0,0 +1,70 @@
+from django.conf import settings
+
+from curling.lib import API
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from slumber.exceptions import HttpClientError
+
+from ..base.views import error_400, error_405
+
+
+def api():
+    # TODO: add oauth.
+    return API(settings.SOLITUDE_URL)
+
+
+class SolitudeBodyguard(APIView):
+    """
+    API view that proxies a single call downstream to Solitude
+    with some restrictions.
+
+    In other words, it's like Solitude's bodyguard because
+    Solitude is firewalled off from direct Internet connections.
+    """
+    # A list of allowed methods, as lowercase Curling properties.
+    # Example: methods=['get', 'post']
+    methods = []
+    # Path to a resource, as a dot-separated Curling attribute path.
+    # Example:
+    # GET /services/status/
+    # would be:
+    # resource = 'services.status'
+    resource = None
+
+    def _api_request(self, method, *args, **kw):
+        if method.lower() not in self.methods:
+            return error_405()
+
+        # Get the endpoint + method, such as api.services.status.get
+        api_request = getattr(self._resource(), method)
+
+        try:
+            result = api_request(*args, **kw)
+        except HttpClientError, exc:
+            return error_400(exception=exc)
+        return Response(result)
+
+    def _resource(self):
+        """
+        Returns the Curling API resource object, i.e. minus the
+        request method.
+
+        For example, when
+
+            resource = 'services.status'
+
+        this returns
+
+            api().services.status
+        """
+        resource = api()
+        url_paths = self.resource.split('.')
+        while len(url_paths):
+            resource = getattr(resource, url_paths.pop(0))
```

Comment:
there's also api.by_url('/some/random/url') in curling that might work for you: https://github.com/andymckay/curling/blob/master/curling/lib.py#L318